x-pack/plugins/fleet/server/routes/data_streams/handlers.ts

```diff
@@ -40,129 +42,8 @@ interface ESDataStreamInfo {
-        size: 1,
-        index: dataStreamName,
-        sort: {
-          // @ts-expect-error Type '{ 'event.ingested': string; }' is not assignable to type 'string | string[] | undefined'.
-          'event.ingested': 'desc',
-        },
-        _source: false,
-        fields: ['event.ingested'],
-      }),
-      esClient.termsEnum({
-        index: dataStreamName,
-        field: 'data_stream.namespace',
-      }),
-      esClient.termsEnum({
-        index: dataStreamName,
-        field: 'data_stream.dataset',
-      }),
-      esClient.termsEnum({
-        index: dataStreamName,
-        field: 'data_stream.type',
-      }),
-    ]);
-
-  const maxIngested = new Date(
-    maxEventIngestedResponse.hits.hits[0]?.fields!['event.ingested']
-  ).getTime();
-
-  const namespace = namespaceResponse.terms[0] ?? '';
-  const dataset = datasetResponse.terms[0] ?? '';
-  const type = typeResponse.terms[0] ?? '';
-
-  return {
-    maxIngested,
-    namespace,
-    dataset,
-    type,
-  };
-}
-
-async function getMetadataFromAggregations({
-  dataStreamName,
-  esClient,
-}: {
-  dataStreamName: string;
-  esClient: ElasticsearchClient;
-}) {
-  // Query backing indices to extract data stream dataset, namespace, and type values
-  const { aggregations: dataStreamAggs } = await esClient.search({
-    index: dataStreamName,
-    body: {
-      size: 0,
-      query: {
-        bool: {
-          filter: [
-            {
-              exists: {
-                field: 'data_stream.namespace',
-              },
-            },
-            {
-              exists: {
-                field: 'data_stream.dataset',
-              },
-            },
-          ],
-        },
-      },
-      aggs: {
-        maxIngestedTimestamp: {
-          max: {
-            field: 'event.ingested',
-          },
-        },
-        dataset: {
-          terms: {
-            field: 'data_stream.dataset',
-            size: 1,
-          },
-        },
-        namespace: {
-          terms: {
-            field: 'data_stream.namespace',
-            size: 1,
-          },
-        },
-        type: {
-          terms: {
-            field: 'data_stream.type',
-            size: 1,
-          },
-        },
-      },
-    },
-  });
-
-  const { maxIngestedTimestamp } = dataStreamAggs as Record<
-    string,
-    estypes.AggregationsRateAggregate
-  >;
-  const { dataset, namespace, type } = dataStreamAggs as Record<
-    string,
-    estypes.AggregationsMultiBucketAggregateBase<{ key?: string; value?: number }>
-  >;
-
-  const maxIngested = maxIngestedTimestamp?.value;
-
-  return {
-    maxIngested,
-    dataset: (dataset.buckets as Array<{ key?: string; value?: number }>)[0]?.key || '',
-    namespace: (namespace.buckets as Array<{ key?: string; value?: number }>)[0]?.key || '',
-    type: (type.buckets as Array<{ key?: string; value?: number }>)[0]?.key || '',
-  };
-}
+const hasASingleBucket = (buckets: Array<string | undefined>): buckets is [string] =>
```

Comment:
I'm not sure if this adds a lot of value over e.g. `serviceNames?.length > 0`?

Reply:
Fixed in 3447de71ac8aa43a5bb2c1359a04b58a956a07ec